geode-core/src/main/java/org/apache/geode/internal/cache/map/RegionMapPut.java

```diff
@@ -430,6 +432,16 @@ private boolean checkCreatePreconditions() {
     return true;
   }
 
+  private boolean isSameValueAlreadyInCacheForPutIfAbsent(Object retainedValue) {
```

Comment:
In this method add a local: Object rawNewValue = getEvent().getRawNewValue();
so that getRawNewValue() will only be called once.